### PR TITLE
Prism Sample

### PR DESCRIPTION
**Pull request by KasiVishwa-RV**

### CODE REVIEW

File: PrismSampleApp/PrismSampleApp/App.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            containerRegistry.RegisterSingleton<IAppInfo, AppInfoImplementation>();
+
+            containerRegistry.RegisterForNavigation<NavigationPage>();
+            containerRegistry.RegisterForNavigation<MainPage, MainPageViewModel>();
+            containerRegistry.RegisterForNavigation<ViewA, ViewAViewModel>();
+            containerRegistry.RegisterForNavigation<ViewB, ViewBViewModel>();
+            containerRegistry.RegisterForNavigation<ViewC, ViewCViewModel>();
+            containerRegistry.RegisterForNavigation<ViewModelLocator, ViewModelLocatorViewModel>();
+            containerRegistry.RegisterForNavigation<Login_Page, Login_PageViewModel>();
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/App.xaml.cs]: Don't use underscore in class name.

File: PrismSampleApp/PrismSampleApp/ViewModels/Login_PageViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        private DelegateCommand _navigateCommand;
+        public DelegateCommand NavigateCommand =>
+            _navigateCommand ?? (_navigateCommand = new DelegateCommand(ExecuteNavigateCommand));
+
+        //private string _userName;
+        //public string UserName
+        //{
+        //    get
+        //    {
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/ViewModels/Login_PageViewModel.cs]: Remove all commented codes.

File: PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            _navigateCommandB ?? (_navigateCommandB = new DelegateCommand(ExecuteNavigateCommandB));
+
+
+        private DelegateCommand _navigateCommandC;
+        public DelegateCommand NavigateCommandC =>
+            _navigateCommandC ?? (_navigateCommandC = new DelegateCommand(ExecuteNavigateCommandC));
+
+        async void ExecuteNavigateCommandA()
+        {
+            var p = new NavigationParameters();
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs]: Name the variable properly.

File: PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+            var q = new NavigationParameters();
+            q.Add("title", "Hello from MainPage");
+
+            await _navigationService.NavigateAsync("ViewB", q);
+        }
+        async void ExecuteNavigateCommandC()
+        {
+
+            var r = new NavigationParameters();
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs]: Name the variable properly everywhere.

File: PrismSampleApp/PrismSampleApp/ViewModels/IViewAViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace PrismSampleApp.ViewModels
+{
+    internal interface IViewAViewModel
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/ViewModels/IViewAViewModel.cs]: Why is this needed? Remove it if not used.

File: PrismSampleApp/PrismSampleApp/ViewModels/ViewBViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        private readonly INavigationService _navigationService;
+        public DelegateCommand NavigateCommandC =>
+            _navigateCommand ?? (_navigateCommand = new DelegateCommand(ExecuteNavigateCommandC));
+
+        async void ExecuteNavigateCommandC()
+        {
+            await _navigationService.NavigateAsync("ViewC");
+        }
+        private DelegateCommand _navigateCommandMainPage;
+        public DelegateCommand GoHomeCommand =>
```
Comment (Jayaram-ces) [inline PrismSampleApp/PrismSampleApp/ViewModels/ViewBViewModel.cs:42]: Use the Delegate command as you used it in Xamarin sample.

**Review by Jayaram-ces — CHANGES_REQUESTED** [review]
- Modify the same prism app to the Contacts app.
- Two simple pages 
   1. Add contact.
   2. View contact list.
- Have two buttons (Add contact & Contacts). When you click Add contact it should save the contact you entered to the list and when you click "Contacts" it should navigate to View contact list page with the current contact list that you entered on the previous page.
- No SQLite / local DB required.

File: PrismSampleApp/PrismSampleApp/App.xaml.cs (hunk 1 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/App.xaml.cs]: Removed the underscore.

File: PrismSampleApp/PrismSampleApp/ViewModels/Login_PageViewModel.cs (hunk 2 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/ViewModels/Login_PageViewModel.cs]: removed the commented codes.

File: PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs (hunk 3 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs]: Variables have been named properly.

File: PrismSampleApp/PrismSampleApp/ViewModels/IViewAViewModel.cs (hunk 5 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/ViewModels/IViewAViewModel.cs]: Variables have been named properly.

File: PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs (hunk 4 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs]: Variables have been named properly.

File: PrismSampleApp/PrismSampleApp/ViewModels/ViewBViewModel.cs (hunk 6 above)
Comment (KasiVishwa-RV) [reply PrismSampleApp/PrismSampleApp/ViewModels/ViewBViewModel.cs:42]: Made changes in the delegate command
